src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -80,8 +80,7 @@ struct sol_http_client_connection {
     struct sol_buffer buffer;
     struct sol_vector watches;
     struct sol_http_params response_params;
-
-    void (*cb)(void *data, const struct sol_http_client_connection *connection, struct sol_http_response *response);
+    struct sol_http_request_interface *interface;
```

Comment:
nah, remove the "*", keep it as a struct instead a pointer to a struct, you're always allocating it and thus you can avoid doing that.